great_expectations_provider/operators/great_expectations.py

```diff
@@ -247,11 +248,21 @@ def make_connection_configuration(self) -> Dict[str, str]:
             odbc_connector = ""
             if conn_type in ("redshift", "postgres"):
                 odbc_connector = "postgresql+psycopg2"
+                database_name = self.schema
             elif conn_type == "mysql":
                 odbc_connector = "mysql"
-            else:
+                database_name = self.schema
+            elif conn_type == "mssql":
                 odbc_connector = "mssql+pyodbc"
-            uri_string = f"{odbc_connector}://{self.conn.login}:{self.conn.password}@{self.conn.host}:{self.conn.port}/{self.schema}"  # noqa
+                ms_driver = self.conn.extra_dejson.get("driver") or "ODBC Driver 17 for SQL Server"
```

Comment:
nitpick: not yet tested myself, but I think the following should work 🤔

```suggestion
                ms_driver = self.conn.extra_dejson.get("driver", "ODBC Driver 17 for SQL Server")
```